src/ctap/data_formats.rs

```diff
@@ -807,7 +758,7 @@ fn extract_map(
     }
 }
 
-pub(super) fn read_bool(cbor_value: &cbor::Value) -> Result<bool, Ctap2StatusCode> {
+pub(super) fn extract_bool(cbor_value: cbor::Value) -> Result<bool, Ctap2StatusCode> {
```

Comment:
You can make this function (and all functions that return a type implementing `Copy`) take `impl Borrow<cbor::Value>` then use `match cbor_value.borrow()`. This would permit to avoid some `.clone()` I've seen above.

Reply:
The benefit of doing so is small (1) and introduces an inconvenience in tests (2).

1. We should not have to `extract_bool` on a reference. The particular existing case may disappear depending on the chosen API on `Extensions`. In particular if `Extensions` become a real Rust struct like others in this file.

2. The pattern `extract_bool(cbor_int!(x))` doesn't work well because `cbor_int!` is polymorphic in its return type while `extract_bool` would be polymorphic in its argument. So the type system can't decide which type to use. One would need to use a monomorph version  of `extract_bool` in the test.